### ISSUE TITLE
Crash in bParking processing

### ISSUE THREAD

**Issue by fabiocos** [cms-sw/cmssw#26154]
A problem with bParking code has been reported by PdmV. This issue has been initially observed during CRAB-based private tests, and were reproducible when running in CRAB but not when running locally.  The problem is:

```
----- Begin Fatal Exception 11-Mar-2019 20:27:43 UTC-----------------------
An exception of category 'FatalRootError' occurred while
[0] Constructing the EventProcessor
[1] Constructing module: class=LowPtGsfElectronIDProducer label='lowPtGsfElectronID'
Additional Info:
[a] Fatal Root Error: @SUB=TXMLEngine::ParseFile
Unexpected end of xml file
----- End Fatal Exception -------------------------------------------------
```

initially reported in https://hypernews.cern.ch/HyperNews/CMS/get/computing-tools/4440.html and https://indico.cern.ch/event/790963/contributions/3306140 .
It looks present in 10_2_X, not in cycles after 10_3_X, starting with merges on top of 10_2_9.

**Comment by fabiocos** [comment]
@prebello could you please point us to a failing workflow/initial settings?

**Comment by fabiocos** [comment]
@mverzett @bainbrid FYI

**Comment by prebello** [comment]
> @prebello could you please point us to a failing workflow/initial settings?

sure for instance
full set of injected relvals are at
https://dmytro.web.cern.ch/dmytro/cmsprodmon/workflows.php?campaign=CMSSW_10_2_12__fullsim_noPU_2018_bParking-1552246543

details for each step can be found at 
https://cmsweb.cern.ch/reqmgr2/fetch?rid=prebello_RVCMSSW_10_2_12SingleElectronPt1000_190310_203726_6599
clicking on the respective step and then "Raw Data" tab

does it help? any further detail?

**Comment by davidlange6** [comment]
previously reported errors were due to special characters in the file (it seems)

The file read here is
RunII_Autumn18_LowPtElectrons_mva_id.xml.gz

is that correct?

**Comment by mverzett** [comment]
@davidlange6 yes, the file is the right one.

> previously reported errors were due to special characters in the file (it seems)

  * Out of curiosity, how can you tell?
  * Why it works locally but not remotely?
  * How do we clean the xml from these characters?

**Comment by davidlange6** [comment]
> On Mar 12, 2019, at 5:20 PM, Mauro Verzetti <notifications@github.com> wrote:
> 
> @davidlange6 yes, the file is the right one.
> 
> previously reported errors were due to special characters in the file (it seems)
> 
> 	• Out of curiosity, how can you tell?
> 	• Why it works locally but not remotely?
> 	• How do we clean the xml from these characters?

sorry I mean previous times when this same error was reported.. 

I don't actually see special characters in this file (yet), so past experience may have nothing to do with the current problem.

**Comment by bainbrid** [comment]
Note that we had to execute the following command:

xmllint --format BDT.xml > BDT_formatted.xml

before we were able to successfully read the files with the TXMLEngine. 

Also we used gzip, as you can see from the file name suffixes. 

So perhaps there are some issues / inconsistencies related to these two executables? 

Rob

**Comment by davidlange6** [comment]
probably good to see if there is better success using an unzipped file. No reason why there should be but..

otherwise, I'm lost about which releases have or have not worked in the various crab tests. Can someone be specific (with three digit release names).

**Comment by guitargeek** [comment]
This seems to be the reoccurrence of https://github.com/cms-sw/cmssw/issues/24395, which got fixed by https://github.com/cms-sw/cmssw/pull/24432 in 10_3_X. Since you are using 10_2_X here, the weight file is just too large I think.

Indeed, when I faced this problem half a year ago it also magically worked locally but not on CRAB...

**Comment by mverzett** [comment]
@guitargeek thanks for the explanation! Indeed it makes sense as the weight files are quite large.
@perrotta @slava77 @fabiocos any chance we can backport #24432 or violates some rules? Any other ideas for a workaround?

**Comment by guitargeek** [comment]
That would of course be quite a big backport, other ideas would be really convenient! In case we want to keep the backport option open however, there is also a cmsdist backport which would have to be made for tinyxml2, analogue to here https://github.com/cms-sw/cmsdist/pull/4433 (except that tinyxml1 should be just kept in the backported case). To make this list of references related to #24432 complete, I should also mention the external PR that it depended on: https://github.com/cms-sw/cmsdist/pull/4332

**Comment by perrotta** [comment]
Given the review comments posted for #24432 (see https://github.com/cms-sw/cmssw/pull/24432#issuecomment-425705984) I see no counterindication in principle in backporting that PR. It must be noted however, as also pointed out by @guitargeek right above. that it would be quite a big backport, it would act on parts of the code which have been touched and modified independently in the meanwhile, and therefore both the backport into a production release and its review should be done and validated carefully. As such, I may expect some non negligible further delay in the start of the bParking reco production.

Moreover (just to keep here some record of it) one has to backport and integrate also the later fix #25085.

If I read correctly the issue #24395, there was a suggestion there of acting on the ```xmlenginebuffersize``` limit (see https://github.com/cms-sw/cmssw/issues/24395#issuecomment-425704753): was that possibility ever explored, as a possible faster alternative?

**Comment by guitargeek** [comment]
I personally never explored that option. But I think it must be somehow more complicated than just changing the limit, because there is also this other strange effect of it working locally but not on CRAB.

What I can do is to explore how easy a backport of #24432 + #25085 would be (i.e. how many conflicts would arise).

**Comment by bainbrid** [comment]
My feeling is that the back port, while a non-negligible effort, is presumably the most sensible way to proceed. So if @guitargeek <https://github.com/guitargeek> can create the PR, then once all the code-checks, etc are done, myself and @mverzett can use the PR branch to test locally and on CRAB to ensure it does the job. Presumably if the problem with CRAB goes away, then one can only hope the newly discovered issues with wmagent also go away ...?

**Comment by guitargeek** [comment]
This is the backport for cmsdist, cmssw will follow.

**Comment by mverzett** [comment]
@bainbrid at the same time we could test the increase of ```xmlenginebuffersize``` with the same assumption that if it works on crab works for pdmV as well. 
@guitargeek is it an environmental variable? a global variable?

**Comment by bainbrid** [comment]
Yes, sounds good - thanks!

**Comment by guitargeek** [comment]
Yes, that's good we try things in parallel. I have no idea how to change the `xmlenginebuffersize` though.

**Comment by mverzett** [comment]
Apparently the buffer size was **hardcoded** in the TMVA code until [this august](https://github.com/root-project/root/commit/9cee1571f1ee952a6c00e4f519e6adc72a17b927). I guess that the backport is the **only** solution

**Comment by mverzett** [comment]
Still, is a mystery how locally it works though

**Comment by guitargeek** [comment]
I'm theoretically done with the backport, but I have real trouble compiling and testing today! Both on lxplus and the machines of by lab I always get `/bin/sh: line 1: 294961 Segmentation fault      (core dumped) edmWriteConfigs -p blabla.so` Is that a known error?

**Comment by mverzett** [comment]
Probably naive question: Could the problem lie [here](https://github.com/cms-sw/cmssw/blob/CMSSW_10_2_X/CommonTools/Utils/src/TMVAZipReader.cc#L80-L84)? Given we have two BDT models being instantiated sequentially in the same module and the comment on security of the line above the quoted ones?

**Comment by guitargeek** [comment]
What do you mean by "have two BDT models being instantiated sequentially in the same module"? There are modules in production for quite some time now which instantiate dozens of GBRForests, read from gzipped files right now. What does the low pt reconstruction do different with respect to what already exists? But honestly I don't quite understand the comment you linked anyway :)

**Comment by Dr15Jones** [comment]
There is a potential problem here
https://github.com/cms-sw/cmssw/blob/CMSSW_10_2_X/CommonTools/Utils/src/TMVAZipReader.cc#L81

`mkstemp` can fail, in which case it returns -1. That is never checked. If we had a failure on mulitple jobs running on a site, they would all try to open `/tmp/tmva.XXXXXX` and write into it simultaneously. Then when TMVA tries to read it, it could be changing while doing the read.

@bbockelm could that be causing the problems we see?

**Comment by bbockelm** [comment]
Nope, that’s not the cause.

1.  All production jobs run in separate containers with unique /tmp.
2.  mkstemp only returns successfully if it has created a new unique file.
3.  The XXXXXX is replaced with randomly generated characters.  That’s why the file name is declared char* and not const char*.

**Comment by Dr15Jones** [comment]
@bbockelm if mkstemp fails, the man page says the value for 'template' is undefined or unchanged, depending on the failure.

**Comment by Dr15Jones** [comment]
Another potential problem in the code is
https://github.com/cms-sw/cmssw/blob/CMSSW_10_2_X/CommonTools/Utils/src/TMVAZipReader.cc#L87-L88

the` fputs` do not check for success.

**Comment by davidlange6** [comment]
do we avoid these possible problems with an unzipped file?

**Comment by Dr15Jones** [comment]
> do we avoid these possible problems with an unzipped file?

Yes, since then a temporary file is never written and instead the unzipped files is passed directly to TMVA.

**Comment by slava77** [comment]
@bainbrid @mverzett 
please test with a non-gzipped file.
I guess it should be easy in the crab submission environment

**Comment by mverzett** [comment]
the non gzipped is 120MB, would that be feasible? TMVA is extremely verbose with the XML.

**Comment by slava77** [comment]
> the non gzipped is 120MB, would that be feasible? TMVA is extremely verbose with the XML.

will that prevent local tests? (the sandbox size problem)

**Comment by davidlange6** [comment]
Certainly it’s not a problem once the file is in the release distribution. There are much bigger


David


On 13 Mar 2019, at 17:43, Mauro Verzetti <notifications@github.com<mailto:notifications@github.com>> wrote:


the non gzipped is 120MB, would that be feasible? TMVA is extremely verbose with the XML.

**Comment by guitargeek** [comment]
I don't know if doubling would be enough. If the buffersize it in bytes, then going from 10 MB to 20 MB does not help much I think, when the actual file is of the order of 100 MB.

Meanwhile I'm preparing a "minimal" backport of the tinyxml2 parser to 10_2_X. It will delete almost no code and then both parsers are available, so that we only need to migrate the low-pt electrons.

**Comment by Dr15Jones** [comment]
I may have encountered another problem. I am trying to run the same configuration in CMSSW_10_6_ASAN to see if it could find a problem. As part of that, I hit the exception

```
----- Begin Fatal Exception 13-Mar-2019 19:06:39 CET-----------------------
An exception of category 'FormulaEvaluatorParseError' occurred while
   [0] Constructing the EventProcessor
   [1] Constructing module: class=CTPPSDiamondLocalTrackFitter label='ctppsDiamondLocalTracks'
Exception Message:
While parsing '(TMath::Erf((x-[0]+0.5*[1])/([2]/4)+2)+1)*TMath::Erfc((x-[0]-0.5*[1])/([2]/4)-2)/4' could not parse beyond '(TMath::Erf((x-[0]+0.5*[1])/([2]/4)+2)+1)*'
----- End Fatal Exception -------------------------------------------------
```
And, indeed, the formula evaluator does not now about TMath::Erfc. This may not have been seen in CMSSW_10_2 since the formula evaluator code has had better error handling in later releases.

**Comment by slava77** [comment]
@jan-kaspar the last comment https://github.com/cms-sw/cmssw/issues/26154#issuecomment-472545292 may be of interest to you; perhaps it's worth to open a separate issue on this.

**Comment by jan-kaspar** [comment]
> @jan-kaspar the last comment [#26154 (comment)](https://github.com/cms-sw/cmssw/issues/26154#issuecomment-472545292) may be of interest to you; perhaps it's worth to open a separate issue on this.

Thanks @slava77 ! Forwarding to @forthommel who is more expert on this (diamond) code. Informing also @fabferro .

**Comment by guitargeek** [comment]
Ok, I think i did everything requested.
1. PR with the full backport of the new parser used everywhere: https://github.com/cms-sw/cmssw/pull/26170
2. Smaller PR with the new parser without removing the old one and changing half of cmssw: https://github.com/cms-sw/cmssw/pull/26176

Local matrix tests pass for both, and I think both are good solutions.

**Comment by slava77** [comment]
I ran valgrind on the reco step of wf 136.8501 (```--era Run2_2018,bParking -s RAW2DIGI,L1Reco,RECO,SKIM:SkimBPark,EI,PAT...```).
There are no bad reads after 7 events processed.

As usual for root, there are hundreds of UninitCondition cases, most likely coming out from the TObject implementation like here https://github.com/root-project/root/blob/v6-12-00-patches/core/base/inc/TObject.h#L230. In the past I could easily ignore these only running on a build with root debug symbols where the TObject.h origin is known .

**Comment by bainbrid** [comment]
I provide below my recipe based on the "minimal" instructions kindly provided by @guitargeek in #26176

```
cmsrel CMSSW_10_2_X_2019-03-13-1100
cd CMSSW_10_2_X_2019-03-13-1100/src/
cmsenv
git cms-init
git cms-merge-topic guitargeek:backport_24432_minimal
cd $CMSSW_BASE/src/CommonTools/MVAUtils/src
wget https://raw.githubusercontent.com/leethomason/tinyxml2/master/tinyxml2.cpp
wget https://raw.githubusercontent.com/leethomason/tinyxml2/master/tinyxml2.h
cd $CMSSW_BASE/src/RecoParticleFlow/PFProducer
git clone git@github.com:cms-data/RecoParticleFlow-PFProducer.git
mv data/ data.bak # there is an existing doc directory, which I moved ...?!
mv RecoParticleFlow-PFProducer/ data
cd $CMSSW_BASE/src
scram b -j8
runTheMatrix.py -l 136.8501 -i all --command "-n 1"
```

It builds and runs locally.

@mverzett, please can you point me to a RAW (data?) file I can access locally and via CRAB that we know produces the TXMLEngine error?

@prebello are you able to also test with WMagent?

**Comment by prebello** [comment]
@bainbrid  there is an error while compiling
gmake: *** [config/SCRAM/GMake/Makefile.rules:1673: tmp/slc6_amd64_gcc700/src/RecoHI/HiTracking/src/RecoHIHiTracking/HIProtoTrackFilter.cc.o] Segmentation fault

**Comment by prebello** [comment]
@bainbrid  I could not compile your receipt. 
Many errors while trying the compilation, as

gmake: *** [config/SCRAM/GMake/Makefile.rules:1673: tmp/slc6_amd64_gcc700/src/RecoHI/HiTracking/src/RecoHIHiTracking/HIProtoTrackFilter.cc.o] Segmentation fault

****WARNING: Invalid tool tinyxml2. Please fix src/CommonTools/MVAUtils/BuildFile.xml file
/bin/sh: line 1:  9473 Segmentation fault      /bin/echo ">> Compiling edm plugin /afs/cern.ch/work/p/prebello/public/PDMV/Bparking/relval/CMSSW_10_2_X_2019-03-13-1100/src/PhysicsTools/PatAlgos/plugins/PATMETProducer.cc "
gmake: *** [config/SCRAM/GMake/Makefile.rules:1673: tmp/slc6_amd64_gcc700/src/PhysicsTools/PatAlgos/plugins/PhysicsToolsPatAlgos_plugins/PATMETProducer.cc.o] Error 139

gmake: *** [config/SCRAM/GMake/Makefile.rules:1673: tmp/slc6_amd64_gcc700/src/RecoEgamma/EgammaElectronAlgos/src/RecoEgammaEgammaElectronAlgos/EgAmbiguityTools.cc.o] Segmentat

>> Compiling edm plugin /afs/cern.ch/work/p/prebello/public/PDMV/Bparking/relval/CMSSW_10_2_X_2019-03-13-1100/src/PhysicsTools/PatAlgos/plugins/PATLostTracks.cc 
In file included from /cvmfs/cms-ib.cern.ch/nweek-02567/slc6_amd64_gcc700/external/gcc/7.0.0-omkpbe2/include/c++/7.3.1/string:48:0,
                 from /afs/cern.ch/work/p/prebello/public/PDMV/Bparking/relval/CMSSW_10_2_X_2019-03-13-1100/src/PhysicsTools/PatAlgos/plugins/PATLostTracks.cc:1:
/cvmfs/cms-ib.cern.ch/nweek-02567/slc6_amd64_gcc700/external/gcc/7.0.0-omkpbe2/include/c++/7.3.1/bits/stl_function.h:1127:11: error: /cvmfs/cms-ib.cern.ch/nweek-02567/slc6_amd64_gcc700/external/gcc/7.0.0-omkpbe2/include/c++/7.3.1/backward/binders.h: Bad address
 # include <backward/binders.h>

gmake: *** [config/SCRAM/GMake/Makefile.rules:1673: tmp/slc6_amd64_gcc700/src/PhysicsTools/PatAlgos/plugins/PhysicsToolsPatAlgos_plugins/PATLostTracks.cc.o] Error 1

**Comment by guitargeek** [comment]
With the segmentation faults it looks a bit like the situation I had this morning: I tried to build it on a SLC7 machine while CMSSW_10_2_X needs SLC6. I finally understood that because at the beginning of the scram output I saw:

`SCRAM warning: You are trying to compile/build for architecture slc6_amd64_gcc700 on SLC7 OS which might not work.`

Maybe you are in the same situation? Concerning the warning about tinyxml2: you can ignore that one. And I think for a successful build, you also need to change `#include <tinyxml2.h>` in the GBRForestTools [1] to `#include "tinyxml2.h"`. You can do that just manually after following the recipe. Maybe it's not necessary so Rob didn't include it in the recipe, but I think for me it didn't work without it.

Cheers,
Jonas

[1] https://github.com/cms-sw/cmssw/blob/master/CommonTools/MVAUtils/src/GBRForestTools.cc#L10

**Comment by prebello** [comment]
@guitargeek  no, I am connected in slc6 
Developer's area is created for available architecture slc6_amd64_gcc700

**Comment by bainbrid** [comment]
I'm testing again the list of commands, line by line.

I'm using SLC6.


Could you try multiple times the command:
scram b -j8
(Sometimes once is not enough.) 

@guitargeek your PR already includes "tinyxml2.h" - I didn't have to change this ...

**Comment by mverzett** [comment]
AFAIK Crab does not know about integration builds, does the recipe work on 10_2_12 or 10_2_12_patch1?

**Comment by guitargeek** [comment]
The recipe should also work on older releases, but I think crab also supports IBs if you set `allowUndistributedCMSSW = True`

https://twiki.cern.ch/twiki/bin/view/CMSPublic/CRAB3ConfigurationFile

**Comment by bainbrid** [comment]
ok, @prebello , all, just to confirm, I tested line by line and it works fine for me. 
Try on more than one occasion: ```scram b -j8```
It takes a while but it should build.

@guitargeek , to note, FYI:
- I had to move an existing ```RecoParticleFlow/PFProducer/data``` directory (just containing a ```download.url``` file) to make way for the git clone renamed as ```data``` 
- The  ```GBRForestTools.cc``` already has ```"tinyxml2.h"``` in the list of headers, I didn't need to change anything ...
- I also see the warning ```****WARNING: Invalid tool tinyxml2. Please fix src/CommonTools/MVAUtils/BuildFile.xml file.``` message but it doesn't seem to affect anything ...?

**Comment by guitargeek** [comment]
Thank you Rob, I edited my PR description accordingly. Seems I accidentally pushed the changed with the `<>` vs. `""`, but nevermind. Yes, I had the warning too and it can be ignored.

**Comment by mverzett** [comment]
I tested the recipe on crab, one file, 400 events. It works! @prebello can you confirm the same?

**Comment by bainbrid** [comment]
Great!! 

Yes @prebello could you please rest too with WMagent?

If this also works, I’d suggest we are good to proceed with the integration of this PR ...

**Comment by prebello** [comment]
@bainbrid  sorry. busy with lectures marathon since yesterday. I will take care of it today only after 5pm CERN.

**Comment by prebello** [comment]
@bainbrid 
again the error in 10 events local test (lxplus scl6)
%MSG-e TkDetLayers:   TSGFromL2Muon:hltL3NoFiltersNoVtxTrajSeedOIHit@streamBeginRun  16-Mar-2019 03:01:26 CET Run: 1 Stream: 7
 ForwardDiskSectorBuilderFromDet: Trying to build Petal Wedge from Dets at different z positions !! Delta_z = -0.950417
%MSG

 I have injected two relvals in the system to test.

**Comment by slava77** [comment]
On 3/15/19 7:15 PM, Patricia Rebello Teles wrote:
> @bainbrid
> again the error in 10 events local test (lxplus scl6)
> %MSG-e TkDetLayers:
> TSGFromL2Muon:hltL3NoFiltersNoVtxTrajSeedOIHit@streamBeginRun
> 16-Mar-2019 03:01:26 CET Run: 1 Stream: 7
> ForwardDiskSectorBuilderFromDet: Trying to build Petal Wedge from Dets
> at different z positions !! Delta_z = -0.950417
> %MSG
> 

If I'm not mistaken, the above error "Trying to build ..." is showing up
in all data processing since 2017 (or was it 2016).
This is not related to bParking


> I have injected two relvals in the system to test.
> 
> —

**Comment by boudoul** [comment]
I confirm that you can safely ignore this message

**Comment by bainbrid** [comment]
Yes, we’ve seen this since the beginning and we ignore.

**Comment by prebello** [comment]
Colleagues, unfortunately 
---- Begin Fatal Exception 16-Mar-2019 17:15:57 UTC-----------------------
An exception of category 'FatalRootError' occurred while
[0] Constructing the EventProcessor
[1] Constructing module: class=LowPtGsfElectronIDProducer label='lowPtGsfElectronID'
Additional Info:
[a] Fatal Root Error: @SUB=TXMLEngine::ParseFile
Unexpected end of xml file

**Comment by davidlange6** [comment]
how was this test performed?

**Comment by bainbrid** [comment]
@prebello did you manage to successfully build the patch detailed here https://github.com/cms-sw/cmssw/issues/26154#issuecomment-472622069 ? And you still see the Exception with this recipe above?

 Is anyone able to submit jobs using WMagent? Are there instructions we can follow? 

We’ve had the recipe available for 5-6 days, so I’m keen to draw a firm conclusion quickly.

**Comment by mverzett** [comment]
@prebello I am surprised that the error still appears, as the with the PR the XML parsing for the low pt electrons in overhauled and ```TXMLEngine``` is replaced. I am wondering as @bainbrid if you included the recipe he suggested.

**Comment by guitargeek** [comment]
Hi @bainbrid, Slava gave me some feedback on the PR during the weekend, and I took the opportunity to also fix the line with `<tinyxml2.h>`. So now you really have to add to the recipe that it should be replaced in `GBRForestToolsTinyXML.cc` with `"tinyxml2.h"`. That does not explain the error @prebello got however. If the PR was merged, there is no chance the TXMLEngine can throw because it's not used.

**Comment by davidlange6** [comment]
its unlikely that any wmagent test is possible with a custom recipe (eg, outside of a release) - thats why I asked how this test was done..

**Comment by bainbrid** [comment]
Ah ok, I'm not aware of the technical details / limitations involved in such a test. Perhaps this explains why @prebello still sees the error.

@fabiocos, all: given Mauro's test with CRAB appears to mitigate the issue, which is very encouraging, is it at all possible to proceed with reviewing/merging #26176, create a new CMSSW_10_2 release as soon as possible, and try again?

We are on an very tight timeline for the bParking processing and I am worried we will miss our slot.

**Comment by prebello** [comment]
> its unlikely that any wmagent test is possible with a custom recipe (eg, outside of a release) - thats why I asked how this test was done..
> […](#)
> On Mar 18, 2019, at 4:32 AM, Mauro Verzetti ***@***.***> wrote: @prebello I am surprised that the error still appears, as the with the PR the XML parsing for the low pt electrons in overhauled and TXMLEngine is replaced. I am wondering as @bainbrid if you included the recipe he suggested. — You are receiving this because you were mentioned. Reply to this email directly, view it on GitHub, or mute the thread.

exaclty.. I did the test in patch1 release only changing the era accordingly. Indeed if the PR with dedicated changes was not merged, it will not work. I have understood that patch1 was the fixed release.

**Comment by prebello** [comment]
> Ah ok, I'm not aware of the technical details / limitations involved in such a test. Perhaps this explains why @prebello still sees the error.
> 
> @fabiocos, all: given Mauro's test with CRAB appears to mitigate the issue, which is very encouraging, is it at all possible to proceed with reviewing/merging #26176, create a new CMSSW_10_2 release as soon as possible, and try again?
> 
> We are on an very tight timeline for the bParking processing and I am worried we will miss our slot.

@bainbrid one needs the fixes in a dedicated release to make the test properly.

**Comment by prebello** [comment]
> Hi @bainbrid, Slava gave me some feedback on the PR during the weekend, and I took the opportunity to also fix the line with `<tinyxml2.h>`. So now you really have to add to the recipe that it should be replaced in `GBRForestToolsTinyXML.cc` with `"tinyxml2.h"`. That does not explain the error @prebello got however. If the PR was merged, there is no chance the TXMLEngine can throw because it's not used.

@guitargeek  I have done it (changed from <tinyxml2.h> to "tinyxml2.h") to fix the compilation error. Indeed it is not the reason of the issue.

**Comment by bainbrid** [comment]
@prebello @slava77  

Sorry - I'm still confused:
- Are you saying that there exists a CMSSW_10_2_12_patch1 release that contains Jonas's fix? 
- And you used this patch release to test with WMagent?

If yes to both, then the question still remains why you see the exception from the TXMLEngine constuctor when the patch no longer uses this code ...

**Comment by bainbrid** [comment]
Then given the successful CRAB test, perhaps we can proceed straight to a new release (i.e. merge and include this PR) and test again?

**Comment by prebello** [comment]
> @prebello @slava77
> 
> Sorry - I'm still confused:
> 
>     * Are you saying that there exists a CMSSW_10_2_12_patch1 release that contains Jonas's fix?
> 
>     * And you used this patch release to test with WMagent?
> 
> 
> If yes to both, then the question still remains why you see the exception from the TXMLEngine constuctor when the patch no longer uses this code ...

I am not saying that CMSSW_10_2_12_patch1 contains the fix. I have said that "I have guessed that CMSSW_10_2_12_patch1 had the fix" . Yes I have used it.

**Comment by prebello** [comment]
@bainbrid do you think we will have problems in rereco 10-2-12? maybe, as relvals are failing too?
(both are injected using wmagent framework)

the samples are still in assignment-approved, although we have requested the block transfer, and I would like to confirm with you to avoid contacting computing for potential problematic runs.
https://dmytro.web.cern.ch/dmytro/cmsprodmon/workflows.php?campaign=Commissioning2018

FYI @davidlange6

**Comment by fabiocos** [comment]
CMSSW_10_2_12_patch1 has nothing to do with this issue, see the release notes https://cmssdt.cern.ch/SDT/ReleaseNotes/CMSSW_10/CMSSW_10_2_12_patch1.html 
I built it to allow GEN-SIM to move forward without interfering with the fix of the B parking issue.

Yesterday I have privately tested the recipe locally, and it works (i.e. the workflow 136.898 arrives at the end without apparent problems). Having a new release with the various additions for B parking inside looks the best way to move forward, provided the backport is approved

**Comment by bainbrid** [comment]
We need a new CMSSW release if we are to use WMagent successfully.

**Comment by prebello** [comment]
PdmV gladly informs that relvals were announced with 100% success. @fabiocos  

https://dmytro.web.cern.ch/dmytro/cmsprodmon/workflows.php?prep_id=CMSSW_10_2_13__fullsim_noPU_2018_bParking-1553134492-SingleElectronPt1000

https://dmytro.web.cern.ch/dmytro/cmsprodmon/workflows.php?prep_id=CMSSW_10_2_13__fullsim_noPU_2018_bParking-1553134492-SingleElectronPt10

https://dmytro.web.cern.ch/dmytro/cmsprodmon/workflows.php?prep_id=CMSSW_10_2_13__fullsim_noPU_2018_bParking-1553134492-SingleElectronPt35

https://dmytro.web.cern.ch/dmytro/cmsprodmon/workflows.php?prep_id=CMSSW_10_2_13__fullsim_noPU_2018_bParking-1553134492-ZEE_13

**Comment by fabiocos** [comment]
@prebello thank you a lot
@bainbrid @slava77 @perrotta I consider this issue as solved

**Comment by bainbrid** [comment]
Yes thank you @prebello !
